### DISCUSSION TITLE
Train.predict() call in callback raises an error

### DISCUSSION THREAD

**Discussion by whaowhao** [Lightning-AI/pytorch-lightning#10361] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I have having the same error as [here](https://forums.pytorchlightning.ai/t/train-predict-call-in-callback-raises-an-error/1126) where calling trainer.predict inside a custom callback raises error

**Comment by tchaton** [comment]
Dear @whaowhao. 

Using trainer.preditct `on_epoch_end` within a Callback is quite hacky and it fails as it is not used properly.

I would advise you to use Loop Customization instead. Here is an example with KFold: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/loop_examples/kfold.py#L158

Best,
T.C
